Sources/Subs-Auth.php

```diff
@@ -884,15 +884,15 @@ function hash_password($username, $password, $cost = null)
 }
 
 /**
- * Hashes password with salt, this is solely used for cookies.
+ * Hashes password with salt and authentication secret. This is solely used for cookies.
  *
  * @param string $password The password
  * @param string $salt The salt
  * @return string The hashed password
  */
 function hash_salt($password, $salt)
 {
-	return hash('sha512', $password . $salt);
+	return hash_hmac('sha1', sha1($password . $salt), get_auth_secret());
```

Comment:
Why we switch from sha2 to sha1, how got already first issues:
```
Additionally, SHA1 has also been deemed quite vulnerable to collision attacks
which is why all browsers will be removing support for certificates signed with SHA1 by January 2017.
SHA256 however, is currently much more resistant to collision attacks
as it is able to generate a longer hash which is harder to break.
```
https://www.keycdn.com/support/sha1-vs-sha256

Reply:
For simple hashes, yes, SHA1 has been broken. But this is using `hash_hmac()`, not `hash()`. [HMAC](https://en.wikipedia.org/wiki/HMAC) is a fundamentally different kind of algorithm than a simple hash, and it has not been broken. It is still 100% safe to use even MD5 in an HMAC.

Unfortunately, we cannot simply replace `hash()` with `hash_hmac()` everywhere in our code, because an HMAC is only suitable for certain situations—specifically, message authentication. But authenticating cookies is one of those situations, so switching from `hash()` to `hash_hmac()` is a good idea no matter what. 

In SMF 2.0.16 and higher, we are using an SHA1 HMAC for this task. So by using the same here, we will allow cookies set by 2.0.16+ to be accepted by 2.1 just as they are. That means that upgrading from 2.0.x to 2.1 will not require all users of a forum to log in again, which is better from a user perspective.